core/src/main/java/org/apache/gravitino/job/local/LocalJobExecutor.java

```diff
@@ -0,0 +1,300 @@
+import java.io.IOException;
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import org.apache.commons.lang3.tuple.Pair;
+import org.apache.gravitino.connector.job.JobExecutor;
+import org.apache.gravitino.exceptions.NoSuchJobException;
+import org.apache.gravitino.job.JobHandle;
+import org.apache.gravitino.job.JobTemplate;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class LocalJobExecutor implements JobExecutor {
+
+  private static final Logger LOG = LoggerFactory.getLogger(LocalJobExecutor.class);
+
+  private static final String LOCAL_JOB_PREFIX = "local-job-";
+
+  private static final long UNEXPIRED_TIME_IN_MS = -1L;
+
+  private Map<String, String> configs;
+
+  private BlockingQueue<Pair<String, JobTemplate>> waitingQueue;
+
+  private ExecutorService jobExecutorService;
+
+  // The job status map to keep track of the status of each job. In the meantime, the job status
+  // will be stored in the entity store, so we will clean the finished, cancelled and failed jobs
+  // from the map periodically to save the memory.
+  private Map<String, Pair<JobHandle.Status, Long>> jobStatus;
+  private final Object lock = new Object();
+
+  private long jobStatusKeepTimInMs;
+  private ScheduledExecutorService jobStatusCleanupExecutor;
+
+  private volatile boolean finished = false;
+
+  private Map<String, Process> runningProcesses;
+
+  @Override
+  public void initialize(Map<String, String> configs) {
+    this.configs = configs;
+
+    int waitingQueueSize =
+        configs.containsKey(WAITING_QUEUE_SIZE)
+            ? Integer.parseInt(configs.get(WAITING_QUEUE_SIZE))
+            : DEFAULT_WAITING_QUEUE_SIZE;
+    Preconditions.checkArgument(
+        waitingQueueSize > 0,
+        "Waiting queue size must be greater than 0, but got: %s",
+        waitingQueueSize);
+
+    this.waitingQueue = new LinkedBlockingQueue<>(waitingQueueSize);
+
+    int maxRunningJobs =
+        configs.containsKey(MAX_RUNNING_JOBS)
+            ? Integer.parseInt(configs.get(MAX_RUNNING_JOBS))
+            : DEFAULT_MAX_RUNNING_JOBS;
+    Preconditions.checkArgument(
+        maxRunningJobs > 0, "Max running jobs must be greater than 0, but got: %s", maxRunningJobs);
+
+    this.jobExecutorService =
+        Executors.newFixedThreadPool(
+            maxRunningJobs,
+            runnable -> {
+              Thread thread = new Thread(runnable);
+              thread.setName("LocalJobExecutor-" + thread.getId());
+              thread.setDaemon(true);
+              return thread;
+            });
+    // Start the job executor service to run jobs
+    for (int i = 0; i < maxRunningJobs; i++) {
+      jobExecutorService.submit(this::runJob);
```

Comment:
In this way, there will always `maxRunningJobs` running threads even there is no jobs at all.

I suggest we use a thread to monitor the queue and use the thread pool the handle the job

// one thread

while (isRunnning) {
     try {
        jobPair = waitingQueue.poll(3000, TimeUnit.MILLISECONDS);
        if (jobPair == null) {
          // If no job is available, continue to the next iteration
          continue;
        }
         
        threadpool.submit(jobPair)
      }
}